Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -278,7 +278,32 @@ struct ParametersAndReturnValidator {
             guard let signature = mixin.getValueIfPresent(for: SymbolGraph.Symbol.FunctionSignature.self) else {
                 continue
             }
-            signatures[DocumentationDataVariantsTrait(for: selector)] = signature
+            
+            let trait = DocumentationDataVariantsTrait(for: selector)
+            // Check if we've already encountered a different signature for another platform
+            guard var existing = signatures.removeValue(forKey: trait) else {
+                signatures[trait] = signature
+                continue
+            }
+            
+            // An internal helper function that compares parameter names
+            func hasSameNames(_ lhs: SymbolGraph.Symbol.FunctionSignature.FunctionParameter, _ rhs: SymbolGraph.Symbol.FunctionSignature.FunctionParameter) -> Bool {
+                lhs.name == rhs.name && lhs.externalName == rhs.externalName
+            }
+            // If the two signatures have different parameters, add any missing parameters.
+            // This allows for documenting parameters that are only available on some platforms.
+            if !existing.parameters.elementsEqual(signature.parameters, by: hasSameNames) {
+                for case .insert(offset: let offset, element: let element, _) in signature.parameters.difference(from: existing.parameters, by: hasSameNames) {
+                    existing.parameters.insert(element, at: offset)
+                }
+            }
```

Comment:
I _think_ this `if` statement is not needed because if there's no difference then the array will be empty and the loop will iterate over nothing (which is the desired behaviour)

```suggestion
            for case .insert(offset: let offset, element: let element, _) in signature.parameters.difference(from: existing.parameters, by: hasSameNames) {
                existing.parameters.insert(element, at: offset)
            }
```

Reply:
The extra if statement isn't necessary but it's faster in the common case where the API has the same signature on all platforms.

Here's a chart (that I made just now) of a microbenchmark where I create 1000 fake symbols, that each exist on two platforms and have an increasing number of fake parameters named "AAAAAAA", "BBBBBBB", "CCCCCCC", etc. on both platforms.

<img width="1280" alt="chart" src="https://github.com/swiftlang/swift-docc/assets/1228143/b4d780c2-396a-438f-a3da-ee0673c23555">

You can see that the `equalElements(_:by:)` check makes a fairly fast early exit compared to much more sophisticated `differences(from:by)` computation that finds no differences. As the symbols get more and more parameters this difference is lessened but at fewer than 10 parameters, it's a rather significant difference.

If I update the benchmark to have minor differences between the parameters of each platform for all 1000 fake symbols, the overhead of the redundant `equalElements(_:by:)` check is barely measurable:

<img width="1280" alt="chart" src="https://github.com/swiftlang/swift-docc/assets/1228143/bb5d6910-b218-4e78-b67f-a299f1440383">

Reply:
I'll add a shorter version of this comment to the code so that someone else (including me when I inevitably forget about this) doesn't remove the if-statement in the future.

Reply:
Thank you for the deep-dive, sounds great!